test/test_profile.py

```diff
@@ -1,83 +1,81 @@
+    except CommandExecutionError:
+        pass
+    assert (browser_params[0].profile_archive_dir / "profile.tar.gz").is_file()
+
+
+def test_profile_error(default_params, task_manager_creator):
+    manager_params, browser_params = default_params
+    manager_params.num_browsers = 1
+    browser_params[0].seed_tar = Path("/tmp/NOTREAL")
+    with pytest.raises(ProfileLoadError):
+        task_manager_creator((manager_params, browser_params[:1]))
 
-    @pytest.mark.xfail(run=False)
-    def test_saving(self):
-        manager_params, browser_params = self.get_config()
-        manager = TaskManager(manager_params, browser_params)
-        manager.get("http://example.com")
-        manager.close()
-        assert isfile(join(browser_params[0].profile_archive_dir, "profile.tar.gz"))
-
-    @pytest.mark.xfail(run=False)
-    def test_crash_profile(self):
-        manager_params, browser_params = self.get_config()
-        manager_params.failure_limit = 2
-        manager = TaskManager(manager_params, browser_params)
-        try:
-            manager.get("http://example.com")  # So we have a profile
-            manager.get("example.com")  # Selenium requires scheme prefix
-            manager.get("example.com")  # Selenium requires scheme prefix
-            manager.get("example.com")  # Selenium requires scheme prefix
-            manager.get("example.com")  # Requires two commands to shut down
-        except CommandExecutionError:
-            pass
-        assert isfile(join(browser_params[0].profile_archive_dir, "profile.tar.gz"))
-
-    @pytest.mark.xfail(run=False)
-    def test_profile_error(self):
-        manager_params, browser_params = self.get_config()
-        browser_params[0].seed_tar = "/tmp/NOTREAL"
-        with pytest.raises(ProfileLoadError):
-            TaskManager(manager_params, browser_params)  # noqa
-
-    @pytest.mark.skip(reason="proxy no longer supported, need to update")
-    def test_profile_saved_when_launch_crashes(self):
-        manager_params, browser_params = self.get_config()
-        browser_params[0].proxy = True
-        browser_params[0].save_content = "script"
-        manager = TaskManager(manager_params, browser_params)
-        manager.get("http://example.com")
 
-        # Kill the LevelDBAggregator
-        # This will cause the proxy launch to crash
-        manager.ldb_status_queue.put("DIE")
-        manager.browsers[0]._SPAWN_TIMEOUT = 2  # Have timeout occur quickly
-        manager.browsers[0]._UNSUCCESSFUL_SPAWN_LIMIT = 2  # Quick timeout
-        manager.get("example.com")  # Cause a selenium crash
-
-        # The browser will fail to launch due to the proxy crashes
-        try:
-            manager.get("http://example.com")
-        except CommandExecutionError:
-            pass
-        manager.close()
-        assert isfile(join(browser_params[0].profile_archive_dir, "profile.tar.gz"))
+@pytest.mark.skip(reason="proxy no longer supported, need to update")
```

Comment:
Just a reminder that we also want to figure out how to fix this.

Reply:
Let's track this in https://github.com/mozilla/OpenWPM/issues/789, so that it doesn't delay this PR. I'll add a comment there.